### ISSUE TITLE
some normal using flyd error

### ISSUE THREAD

**Issue by xialvjun** [theadam/react-flyd#5]
There were 4 bugs or something I thought shouldn't be in that way.
1. extra span tag.... Still there till now
2. change a non stream prop of a reactive element won't change it's state.... Have solved.
3 and 4 are some flyd using error:
3. code like

```
{this.state.show ? (
  <div>
     {num.map(n => <span>{n}</span>)}
  </div>
 ) : null}
```

will make wrong after show and hide the div
4. code just like 3 will lead to a memory leaking...

My way to solve 3 and 4 is to end the prop$ rather than on(xxx).end() on unsubscribe, since we use much more `flyd.map` than the original data stream in jsx...

Whatever will you adopt my code, .....  [my gist](https://gist.github.com/xialvjun/341ec56cceb7bb00cc8455921e875ecd)

Besides, I have changed your package to `preact`---- [preact-flyd](https://github.com/xialvjun/preact-flyd)
Thanks for your idea.

**Comment by theadam** [comment]
Not seeing the first 2 issues in #4.  Did you try with that code?

**Comment by theadam** [comment]
issue 3 and 4 look like issues with react-flyd-class and not with react-flyd.

Ending the passed in prop could be detrimental if other things are using that same stream.  I actually think this is an issue because react-flyd-class calls `end` but does not pass `true` to it.

https://github.com/theadam/react-flyd-class/blob/master/src/createReactiveClass.js#L77

**Comment by theadam** [comment]
https://github.com/theadam/react-flyd-class/pull/6

That PR will fix the issue, and this will be fixed here when `react-flyd` is updated to a newly released version of `react-flyd-class`

I have added a currently failing example to the example code in #4. It will start passing when a new version of react-flyd-class is released.

**Comment by theadam** [comment]
These issues are now fixed with v0.1.1 of react-flyd-class, and will be fixed in react-flyd with the merge and release of #4.

**Comment by theadam** [comment]
Fixed in v0.2.1
